src/canmatrix/cmjson.py

```diff
@@ -101,21 +101,21 @@ def dump(db, f, **options):
                               "signals": signals}
             frame_attributes = {}
             for frame_info in additionalFrameColums:  # Look for additional Frame Attributes
-                if hasattr(frame, frame_info):
-                    frame_attributes[frame_info] = getattr(frame, frame_info)
+                if frame.attribute(frame_info):  # does the attribute exist?
+                    frame_attributes[frame_info] = frame.attribute(frame_info)
             if frame_attributes:  # only add Attributes if there are any
                 symbolic_frame["attributes"] = frame_attributes
             exportArray.append(symbolic_frame)
-    else:  # exportall
+    else:  # exportAll
         for frame in db.frames:
             frameattribs = {}
             for attribute in db.frameDefines:
-                frameattribs[attribute] = frame.attribute(db,attribute)
+                frameattribs[attribute] = frame.attribute(attribute, db)
```

Comment:
Let's specify by name to avoid this hazard in the future.  Looks like this applies to most of the rest of the diff below.

Reply:
Uff, I realized the PyCharm has refactored also calls of another classes. Python has certain weaknesses :-(
Checked, reverted unwanted changes. For Frame use named attribute.